core/modules/replica/DatabaseMySQL.h

```diff
@@ -51,410 +58,122 @@ namespace database {
- * by some methods when a query is supposed to return at least one row.
- */
-class EmptyResultSetError
-    :   public Error {
-
-public:
-
-    /**
-     * The normal constructor
+     * @note
+     *    MySQL auto-commits are disabled
      *
-     * @param what - reason for the exceptioon
-     */
-    explicit EmptyResultSetError(std::string const& what)
-        :   Error(what) {
-    }
-};
-
-/**
- * Struvct ConnectionParams encapsulates connection parameters to a MySQL server
- */
-struct ConnectionParams {
-    /// The DNS name or IP address of a machine where the database
-    /// server runs
-    std::string host;
-
-    /// The port number of the MySQL service
-    uint16_t port;
-
-    /// The name of a database user
-    std::string user;
-
-    /// The database password
-    std::string password;
-
-    /// The name of a database to be set upon the connection
-    std::string database;
-
-    /**
-     * The factory method will return an instance of this structure initialized
-     * by values of parameters found in the input encoded string. The string is
-     * expected to have the following syntax:
+     * @note
+     *    MySQL automatic re-connects are not allowed because this Connector class
+     *    implements its own protocol for reconnects (when allowed)
      *
-     *   mysql://[user][:password]@[host][:port][/database]
      *
-     * NOTES ON THE SYNTAX:
-     * 1) all keywords are madatory
-     * 2) the corresponding values for for all but the database are optional
-     * 3) default values for other parameters (if missing in the string) will be assumed.
+     * Here is an example of using this method to establish a connection:
+     * @code
      *
-     * @param params          - connection parameters packed into a string
-     * @param defaultHost     - default value for a host name
-     * @param defaultPort     - default port number
-     * @param defaultUser     - default value for a database user account
-     * @param defaultPassword - default value for a database user account
+     *    BlockPost delayBetweenReconnects(1000,2000);
+     *    ConnectionParams params = ...;
+     *    Connection::Ptr conn;
+     *    do {
+     *        try {
+     *            conn = Connection::open(params);
+     *            
+     *        } catch (ConnectError const& ex) {
+     *            cerr << "connection attempt failed: " << ex.what << endl;
+     *            delayBetweenReconnects.wait();
+     *            cerr << "reconnecting..." << endl;
+     *            
+     *        } catch (ConnectTimeout const& ex) {
+     *            cerr << "connection attempt expired after: " << ex.timeoutSec() << " seconds "
+     *                 << "due to: " << ex.what << endl;
+     *            throw;
+     *            
+     *        } catch (Error const& ex) {
+     *            cerr << "connection attempt failed: " << ex.what << endl;
+     *            throw;
+     *        }
+     *   } while (nullptr != conn);
```

Comment:
This won't matter here because of reconnects. If an exception happens then `Connection::open` never has a chance to return a pointer. Hence the previously (before the loop pointer `Connection::Ptr` stays equal to `nullptr`). The only chance this changes is when `Connection::open` successfully finishes.